examples/web-app/src/components/Home/Home.js

```diff
@@ -0,0 +1,196 @@
+import React, { useState, useEffect } from "react";
+import { UserFactory } from "nightfall-sdk";
+import "./Home.css";
+
+export default function User() {
+  const [userAddress, setUserAddress] = useState();
+  const [userNightfallAddress, setUserNightfallAddress] = useState();
+  const [userRecipient, setUserRecipient] = useState();
+  const [nightfallMnemonic, setNightfallMnemonic] = useState();
+  const [clientApiUrl, setClientApiUrl] = useState();
+  const [nightfallBalances, setNightfallBalances] = useState(0);
+  const [nightfallMnemonicRecepient, setNightfallMnemonicRecepient] =
+    useState(null);
+
+  const tokenContractAddress = "0xa8473bEF03cBE50229a39718CBDC1fdee2F26b1a";
+  const tokenErcStandard = "ERC20";
+  const txValue = "0.0001";
+
+  let createUserFirstTime = async () => {
```

Comment:
This function will be called each time the `User` component re-renders. It means, pretty often. We can probably move this function to the upperScope, or do some meoization to prevent this function from being called, if it was called before.

Reply:
The purpose of the `createUser` function is to mainly to update the users Nightfall balances. To execute the checkBalances function we need an instance of User so a user needs to be created anyway. Since I am not storing the entire object in localStorage.